projects/control-service/projects/pipelines_control_service/src/integration-test/java/com/vmware/taurus/datajobs/it/common/JobExecutionUtil.java

```diff
@@ -161,43 +162,46 @@ public static void testDataJobExecutionRead(
       String username,
       MockMvc mockMvc) {
 
-    com.vmware.taurus.controlplane.model.data.DataJobExecution[] dataJobExecution =
-        new com.vmware.taurus.controlplane.model.data.DataJobExecution[1];
-
-    await()
-        .atMost(5, TimeUnit.MINUTES)
-        .with()
-        .pollInterval(15, TimeUnit.SECONDS)
-        .until(
-            () -> {
-              String dataJobExecutionReadUrl =
-                  String.format(
-                      "/data-jobs/for-team/%s/jobs/%s/executions/%s",
-                      teamName, jobName, executionId);
-              MvcResult dataJobExecutionResult =
-                  mockMvc
-                      .perform(
-                          get(dataJobExecutionReadUrl)
-                              .with(user(username))
-                              .contentType(MediaType.APPLICATION_JSON))
-                      .andExpect(status().isOk())
-                      .andReturn();
-
-              dataJobExecution[0] =
-                  objectMapper.readValue(
-                      dataJobExecutionResult.getResponse().getContentAsString(),
-                      com.vmware.taurus.controlplane.model.data.DataJobExecution.class);
-              if (dataJobExecution[0] == null) {
-                log.info("No response from server");
-              } else {
-                log.info("Response from server  " + dataJobExecution[0].getStatus());
-              }
-              return dataJobExecution[0] != null
-                  && executionStatus.equals(dataJobExecution[0].getStatus());
-            });
-
-    assertDataJobExecutionValid(
-        executionId, executionStatus, opId, dataJobExecution[0], jobName, username);
+    Callable<com.vmware.taurus.controlplane.model.data.DataJobExecution> dataJobExecutionCallable =
+        () -> {
+          String dataJobExecutionReadUrl =
+              String.format(
+                  "/data-jobs/for-team/%s/jobs/%s/executions/%s", teamName, jobName, executionId);
+          MvcResult dataJobExecutionResult =
+              mockMvc
+                  .perform(
+                      get(dataJobExecutionReadUrl)
+                          .with(user(username))
+                          .contentType(MediaType.APPLICATION_JSON))
+                  .andExpect(status().isOk())
+                  .andReturn();
+
+          return objectMapper.readValue(
+              dataJobExecutionResult.getResponse().getContentAsString(),
+              com.vmware.taurus.controlplane.model.data.DataJobExecution.class);
+        };
+    var result =
+        await()
+            .atMost(11, TimeUnit.MINUTES)
+            .with()
+            .pollInterval(15, TimeUnit.SECONDS)
+            .failFast(
+                () -> {
+                  com.vmware.taurus.controlplane.model.data.DataJobExecution status =
+                      dataJobExecutionCallable.call();
+                  return status != null
+                      && !status
```

Comment:
In my opinion, we should exclude `SUBMITTED` as well.

Reply:
oh nice spot.